idaes/core/util/model_diagnostics.py

```diff
@@ -694,10 +714,152 @@ def find_irreducible_degenerate_sets(self, verbose=True, tee=False):
+        cmeta = getattr(meta, n)[i]
+        valid_range = cmeta.valid_range
+    except ValueError:
+        # Assume no metadata for this property
+        _log.debug(f"No metadata entry for component {component.name}; returning None")
+        valid_range = None
+
+    return valid_range
+
+
+def set_bounds_from_valid_range(component, descend_into=True):
+    """
+    Set bounds on Pyomo components based on valid range recorded in model metadata.
+    WARNING - this function will overwrite any bounds already set on the component/model.
+
+    This function will iterate over component data objects in Blocks and indexed components.
+
+    Args:
+        component: Pyomo component to set bounds on. This can be a Block, Var or Param.
+        descend_into: (optional) Whether to descend into components on child Blocks (default=True)
+
+    Returns:
+         None
+
+    """
+    if component.is_indexed():
+        for k in component:
+            set_bounds_from_valid_range(component[k])
+    elif isinstance(component, _BlockData):
+        for i in component.component_data_objects(
+            ctype=[pyo.Var, pyo.Param], descend_into=descend_into
+        ):
+            set_bounds_from_valid_range(i)
+    elif not hasattr(component, "bounds"):
+        raise TypeError(
+            f"Component {component.name} does not have bounds. Only Vars and Params have bounds."
+        )
+    else:
+        valid_range = get_valid_range_of_component(component)
+
+        if valid_range is None:
+            valid_range = (None, None)
+
+        component.setlb(valid_range[0])
+        component.setub(valid_range[1])
+
+
+def list_components_with_values_outside_valid_range(component, descend_into=True):
+    """
+    Return a list of component objects with values outside the valid range specified in the model
+    metadata.
+
+    This function will iterate over component data objects in Blocks and indexed components.
+
+    Args:
+        component: Pyomo component to search for component outside of range on.
+            This can be a Block, Var or Param.
+        descend_into: (optional) Whether to descend into components on child Blocks (default=True)
+
+    Returns:
+         list of component objects found with values outside the valid range.
+    """
+    comp_list = []
+
+    if component.is_indexed():
+        for k in component:
+            comp_list.extend(
+                list_components_with_values_outside_valid_range(component[k])
+            )
+    elif isinstance(component, _BlockData):
+        for i in component.component_data_objects(
+            ctype=[pyo.Var, pyo.Param], descend_into=descend_into
+        ):
+            comp_list.extend(list_components_with_values_outside_valid_range(i))
+    else:
+        valid_range = get_valid_range_of_component(component)
+
+        if valid_range is not None:
+            cval = pyo.value(component)
+            if cval is not None and (cval < valid_range[0] or cval > valid_range[1]):
```

Comment:
Do you not consider ``None`` to be outside the valid range?

Reply:
I think I did this in case of stale variables (i.e. a case where there was a variable with no initial value that did not appear in any constraints). I would also technically consider `None` to be a different issue (no value versus out of range), but we could include it here if we think it is better.

Reply:
I guess it depends on the use case.  Maybe there is also another check for not set variables.  Intuitively I'd say None is outside the valid range.  I don't think I'm catching the point about stale.  Do you mean if a variable has a value of None, that you want to ignore it because if you've run the problem and it is still None then it is not used and doesn't matter?

Reply:
Yes - I didn't want unused variables cluttering up the output. I was also suggesting there should be a separate check for variables with no value.